core/common/src/main/java/alluxio/util/ConfigurationUtils.java

```diff
@@ -269,6 +270,37 @@ public static boolean jobMasterHostConfigured(AlluxioConfiguration conf) {
         || getJobMasterRpcAddresses(conf).size() > 1;
   }
 
+  /**
+   * Returns a unified message for cases when the master hostname cannot be determined.
+   *
+   * @param serviceName The name of the service that couldn't run. i.e. Alluxio worker, fsadmin
+   *                    shell, etc.
+   * @return a string with the message
+   */
+  public static String getMasterHostNotConfiguredMessage(String serviceName) {
+    return getHostNotConfiguredMessage(serviceName, "master", PropertyKey.MASTER_HOSTNAME,
+        PropertyKey.MASTER_EMBEDDED_JOURNAL_ADDRESSES);
+  }
+
+  /**
+   * Returns a unified message for cases when the job master hostname cannot be determined.
+   *
+   * @param serviceName The name of the service that couldn't run. i.e. Alluxio worker, fsadmin
+   *                    shell, etc.
+   * @return a string with the message
+   */
+  public static String getJobMasterHostNotConfiguredMessage(String serviceName) {
+    return getHostNotConfiguredMessage(serviceName, "job master", PropertyKey.JOB_MASTER_HOSTNAME,
+        PropertyKey.JOB_MASTER_EMBEDDED_JOURNAL_ADDRESSES);
+  }
+
+  private static String getHostNotConfiguredMessage(String serviceName, String masterName,
+      PropertyKey masterHostnameKey, PropertyKey embeddedJournalKey) {
+    return ExceptionMessage.UNABLE_TO_DETERMINE_MASTER_HOSTNAME.getMessage(serviceName, masterName,
```

Comment:
imo it would be less error-prone to put the format string here so that readers can easily line up the parameters with the placeholders. Leaving it up to you

Reply:
I'm going to elect to leave it within `ExceptionMessage`. I like to keep the error messages there